### ISSUE TITLE
[Feature Request] Make Update Fulfilment Status easily accessible

### ISSUE THREAD

**Issue by martbase** [concrete5-community-store/community_store#600]
It'd be nice to have the Update Fulfilment Status select option more prominent and easily accessible in the Orders page/dashboard.

Scenario: I imagine in a busy online shop say a restaurant where there are many orders coming in, viewing orders and adjusting their status as they go through processing and finally shipping is central to the operation of the store.

Should this feature also be developed as an Addon or it'd be better if Community Store improved the UI/UX to streamline processing orders thereby simplifying the workflow instead of having to open each order individually to change its status.

**Comment by Mesuva** [comment]
I agree it's a little hidden towards the bottom of the order page. The challenge is that everyone is processing orders a bit differently, with some not even interested in changing that status - the paid status on the other hand might be more relevant to some.

I'm certainly open to improving the interface though, either through a PR or an example of how it should be.

But this is also a case where a custom built add-on to offer an interface specifically for your use-case might be the way to go. In the EasyPost add-on, I effectively did that - it adds a new dashboard page that lists orders, but with different buttons and info, specifically with the context of managing the shipping of the orders.

The added bonus is that you could use permissions to allow access to only that page by a staff member, if you wanted to stop them from doing other things (like refunding orders).

There's also the API add-on now too, that allows you to externally/programatically fetch order details, and update order statuses, so you could even build something completely outside of the concrete5 site.

**Comment by martbase** [comment]
Great! Thanks for the detailed explanation and ideas. I'll explore both the single page dashboard option and the API integration with an external application.
Had looked at video of the EasyPost add-on and it is quite impressive so will have a look at the code for inspiration on how to implement the dashboard page.

**Comment by martbase** [comment]
> 
> 
> I agree it's a little hidden towards the bottom of the order page. The challenge is that everyone is processing orders a bit differently, with some not even interested in changing that status - the paid status on the other hand might be more relevant to some.
> 
> I'm certainly open to improving the interface though, either through a PR or an example of how it should be.
> 
> But this is also a case where a custom built add-on to offer an interface specifically for your use-case might be the way to go. In the EasyPost add-on, I effectively did that - it adds a new dashboard page that lists orders, but with different buttons and info, specifically with the context of managing the shipping of the orders.
> 
> The added bonus is that you could use permissions to allow access to only that page by a staff member, if you wanted to stop them from doing other things (like refunding orders).
> 
> There's also the API add-on now too, that allows you to externally/programatically fetch order details, and update order statuses, so you could even build something completely outside of the concrete5 site.

Haven't thought this through but this is the idea I had in mind when opening the issue (see attached)

![image](https://user-images.githubusercontent.com/3264387/106306431-1c435a00-626f-11eb-9124-054326cc19c2.png)

**Comment by Mesuva** [comment]
That makes a lot of sense to me. There is also the comment field to consider, but perhaps that's not a concern, as someone can still go into the order to set one.

My only hesitation is the potential for orders to be mixed up, since there's not much detail apart from the name and number shown. When you go into the order you at least have the full order details in front of you, extra things to confirm that you're actually changing the status of the correct order.  

I'm happy to receive a PR for this - I can add it to my list, but not sure when I'd be able to get to it.

**Comment by martbase** [comment]
I agree completely there's potential for mixing up the orders when there are so few details to distinguish between orders. Like I said I haven't thought deeply about this but I feel taking an order through the various fulfillment steps is something that could be useful to a lot of stores especially when coupled with the fulfillment notifications add-on for stores that don't have a way of tracking a customer's order - the customer can be easily be updated on the status of their order as it goes through the processing and shipping stages.
 
Since this status is something that may be changed severally for each individual order there's a need to make it _front and center_. It doesn't have to be a select option that can be changed directly on the page, a pencil icon next to the status would do. When the icon is clicked it would popup a dialog with the most relevant order details (simplified) and provide a control for changing the status thus allowing one to easily confirm the order, update the status and save upon closing the dialog - this dialog box could also have a provision for setting the comment field.
